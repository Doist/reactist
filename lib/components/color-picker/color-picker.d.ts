import * as React from 'react';
import './color-picker.less';
type NamedColor = {
    name: string;
    color: string;
};
declare const COLORS: string[];
type Props = {
    small?: boolean;
    color?: number;
    onChange?: (color: number) => void;
    colorList?: (string | NamedColor)[];
};
declare function ColorPicker({ color, small, onChange, colorList }: Props): React.JSX.Element;
declare namespace ColorPicker {
    var displayName: string;
}
type ColorItemProps = {
    color: string;
    colorIndex: number;
    isActive?: boolean;
    onClick?: (colorIndex: number) => void;
    tooltip?: React.ReactNode;
};
declare function ColorItem({ color, colorIndex, isActive, onClick, tooltip }: ColorItemProps): React.JSX.Element;
declare namespace ColorItem {
    var displayName: string;
}
export { ColorPicker, ColorItem, COLORS };
