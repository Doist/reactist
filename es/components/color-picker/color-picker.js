import * as React from 'react'
import classNames from 'classnames'
import { Dropdown } from '../deprecated-dropdown/dropdown.js'
import { Tooltip } from '../../tooltip/tooltip.js'

const COLORS = [
    '#606060',
    '#4A90E2',
    '#03B3B2',
    '#008299',
    '#82BA00',
    '#D24726',
    '#AC193D',
    '#DC4FAD',
    '#3BD5FB',
    '#74E8D3',
    '#FFCC00',
    '#FB886E',
    '#CCCCCC',
]

const _isNamedColor = (color) => typeof color !== 'string'

const _getColor = (colorList, colorIndex) => {
    const index = colorIndex >= colorList.length ? 0 : colorIndex
    return colorList[index]
}

function ColorPicker({ color = 0, small, onChange, colorList = COLORS }) {
    return /*#__PURE__*/ React.createElement(
        Dropdown.Box,
        {
            right: true,
            className: 'reactist_color_picker',
        },
        /*#__PURE__*/ React.createElement(
            Dropdown.Trigger,
            null,
            (() => {
                const backgroundColor = _getColor(colorList, color)

                return /*#__PURE__*/ React.createElement(
                    'span',
                    {
                        className: classNames('color_trigger', {
                            small,
                        }),
                        style: {
                            backgroundColor: _isNamedColor(backgroundColor)
                                ? backgroundColor.color
                                : backgroundColor,
                        },
                    },
                    /*#__PURE__*/ React.createElement('span', {
                        className: 'color_trigger--inner_ring',
                    }),
                )
            })(),
        ),
        /*#__PURE__*/ React.createElement(
            Dropdown.Body,
            null,
            /*#__PURE__*/ React.createElement(
                'div',
                {
                    className: 'color_options',
                },
                colorList.reduce((items, currentColor, currentIndex) => {
                    items.push(
                        /*#__PURE__*/ React.createElement(ColorItem, {
                            isActive:
                                color >= colorList.length
                                    ? currentIndex === 0
                                    : currentIndex === color,
                            key: currentIndex,
                            color: _isNamedColor(currentColor) ? currentColor.color : currentColor,
                            colorIndex: currentIndex,
                            onClick: onChange,
                            tooltip: _isNamedColor(currentColor) ? currentColor.name : null,
                        }),
                    )
                    return items
                }, []),
            ),
        ),
    )
}

ColorPicker.displayName = 'ColorPicker'

function ColorItem({ color, colorIndex, isActive, onClick, tooltip }) {
    const item = /*#__PURE__*/ React.createElement(
        'span',
        {
            'data-testid': 'reactist-color-item',
            className: 'reactist color_item' + (isActive ? ' active' : ''),
            style: {
                backgroundColor: color,
            },
            onClick: () => (onClick == null ? void 0 : onClick(colorIndex)),
        },
        /*#__PURE__*/ React.createElement('span', {
            className: 'color_item--inner_ring',
        }),
    )
    return tooltip
        ? /*#__PURE__*/ React.createElement(
              Tooltip,
              {
                  content: tooltip,
              },
              item,
          )
        : item
}

ColorItem.displayName = 'ColorItem'

export { COLORS, ColorItem, ColorPicker }
//# sourceMappingURL=color-picker.js.map
