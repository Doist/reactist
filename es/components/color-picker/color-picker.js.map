{"version":3,"file":"color-picker.js","sources":["../../../src/components/color-picker/color-picker.tsx"],"sourcesContent":["import * as React from 'react'\nimport classnames from 'classnames'\n\nimport DeprecatedDropdown from '../deprecated-dropdown'\nimport { Tooltip } from '../../tooltip'\n\nimport './color-picker.less'\n\ntype NamedColor = { name: string; color: string }\n\nconst COLORS = [\n    '#606060',\n    '#4A90E2',\n    '#03B3B2',\n    '#008299',\n    '#82BA00',\n    '#D24726',\n    '#AC193D',\n    '#DC4FAD',\n    '#3BD5FB',\n    '#74E8D3',\n    '#FFCC00',\n    '#FB886E',\n    '#CCCCCC',\n]\n\nconst _isNamedColor = (color: string | NamedColor | undefined): color is NamedColor =>\n    typeof color !== 'string'\n\nconst _getColor = (colorList: (string | NamedColor)[], colorIndex: number) => {\n    const index = colorIndex >= colorList.length ? 0 : colorIndex\n    return colorList[index]\n}\n\ntype Props = {\n    small?: boolean\n    color?: number\n    onChange?: (color: number) => void\n    colorList?: (string | NamedColor)[]\n}\n\nfunction ColorPicker({ color = 0, small, onChange, colorList = COLORS }: Props) {\n    return (\n        <DeprecatedDropdown.Box right className=\"reactist_color_picker\">\n            <DeprecatedDropdown.Trigger>\n                {(() => {\n                    const backgroundColor = _getColor(colorList, color)\n\n                    return (\n                        <span\n                            className={classnames('color_trigger', { small })}\n                            style={{\n                                backgroundColor: _isNamedColor(backgroundColor)\n                                    ? backgroundColor.color\n                                    : backgroundColor,\n                            }}\n                        >\n                            <span className=\"color_trigger--inner_ring\" />\n                        </span>\n                    )\n                })()}\n            </DeprecatedDropdown.Trigger>\n            <DeprecatedDropdown.Body>\n                <div className=\"color_options\">\n                    {colorList.reduce<React.ReactNode[]>((items, currentColor, currentIndex) => {\n                        items.push(\n                            <ColorItem\n                                isActive={\n                                    color >= colorList.length\n                                        ? currentIndex === 0\n                                        : currentIndex === color\n                                }\n                                key={currentIndex}\n                                color={\n                                    _isNamedColor(currentColor) ? currentColor.color : currentColor\n                                }\n                                colorIndex={currentIndex}\n                                onClick={onChange}\n                                tooltip={_isNamedColor(currentColor) ? currentColor.name : null}\n                            />,\n                        )\n                        return items\n                    }, [])}\n                </div>\n            </DeprecatedDropdown.Body>\n        </DeprecatedDropdown.Box>\n    )\n}\nColorPicker.displayName = 'ColorPicker'\n\ntype ColorItemProps = {\n    color: string\n    colorIndex: number\n    isActive?: boolean\n    onClick?: (colorIndex: number) => void\n    tooltip?: React.ReactNode\n}\n\nfunction ColorItem({ color, colorIndex, isActive, onClick, tooltip }: ColorItemProps) {\n    const item = (\n        <span\n            data-testid=\"reactist-color-item\"\n            className={'reactist color_item' + (isActive ? ' active' : '')}\n            style={{ backgroundColor: color }}\n            onClick={() => onClick?.(colorIndex)}\n        >\n            <span className=\"color_item--inner_ring\" />\n        </span>\n    )\n\n    return tooltip ? <Tooltip content={tooltip}>{item}</Tooltip> : item\n}\nColorItem.displayName = 'ColorItem'\n\nexport { ColorPicker, ColorItem, COLORS }\n"],"names":["COLORS","_isNamedColor","color","_getColor","colorList","colorIndex","index","length","ColorPicker","small","onChange","React","createElement","DeprecatedDropdown","Box","right","className","Trigger","backgroundColor","classnames","style","Body","reduce","items","currentColor","currentIndex","push","ColorItem","isActive","key","onClick","tooltip","name","displayName","item","Tooltip","content"],"mappings":";;;;;AAUA,MAAMA,MAAM,GAAG,CACX,SADW,EAEX,SAFW,EAGX,SAHW,EAIX,SAJW,EAKX,SALW,EAMX,SANW,EAOX,SAPW,EAQX,SARW,EASX,SATW,EAUX,SAVW,EAWX,SAXW,EAYX,SAZW,EAaX,SAbW,EAAf;;AAgBA,MAAMC,aAAa,GAAIC,KAAD,IAClB,OAAOA,KAAP,KAAiB,QADrB,CAAA;;AAGA,MAAMC,SAAS,GAAG,CAACC,SAAD,EAAqCC,UAArC,KAA2D;EACzE,MAAMC,KAAK,GAAGD,UAAU,IAAID,SAAS,CAACG,MAAxB,GAAiC,CAAjC,GAAqCF,UAAnD,CAAA;EACA,OAAOD,SAAS,CAACE,KAAD,CAAhB,CAAA;AACH,CAHD,CAAA;;AAYA,SAASE,WAAT,CAAqB;AAAEN,EAAAA,KAAK,GAAG,CAAV;EAAaO,KAAb;EAAoBC,QAApB;AAA8BN,EAAAA,SAAS,GAAGJ,MAAAA;AAA1C,CAArB,EAA8E;AAC1E,EAAA,oBACIW,KAAC,CAAAC,aAAD,CAACC,QAAkB,CAACC,GAApB;AAAwBC,IAAAA,KAAK,EAAA;AAACC,IAAAA,SAAS,EAAC,uBAAA;GAAxC,eACIL,KAAA,CAAAC,aAAA,CAACC,QAAkB,CAACI,OAApB,EACK,IADL,EACK,CAAC,MAAK;AACH,IAAA,MAAMC,eAAe,GAAGf,SAAS,CAACC,SAAD,EAAYF,KAAZ,CAAjC,CAAA;;AAEA,IAAA,oBACIS,KAAA,CAAAC,aAAA,CAAA,MAAA,EAAA;AACII,MAAAA,SAAS,EAAEG,UAAU,CAAC,eAAD,EAAkB;AAAEV,QAAAA,KAAAA;AAAF,OAAlB,CADzB;AAEIW,MAAAA,KAAK,EAAE;QACHF,eAAe,EAAEjB,aAAa,CAACiB,eAAD,CAAb,GACXA,eAAe,CAAChB,KADL,GAEXgB,eAAAA;AAHH,OAAA;AAFX,KAAA,eAQIP,KAAA,CAAAC,aAAA,CAAA,MAAA,EAAA;AAAMI,MAAAA,SAAS,EAAC,2BAAA;AAAhB,KAAA,CARJ,CADJ,CAAA;AAYH,GAfA,GADL,CADJ,eAmBIL,KAAC,CAAAC,aAAD,CAACC,QAAkB,CAACQ,IAApB,EAAwB,IAAxB,eACIV,KAAA,CAAAC,aAAA,CAAA,KAAA,EAAA;AAAKI,IAAAA,SAAS,EAAC,eAAA;GAAf,EACKZ,SAAS,CAACkB,MAAV,CAAoC,CAACC,KAAD,EAAQC,YAAR,EAAsBC,YAAtB,KAAsC;IACvEF,KAAK,CAACG,IAAN,eACIf,KAAA,CAAAC,aAAA,CAACe,SAAD,EAAU;AACNC,MAAAA,QAAQ,EACJ1B,KAAK,IAAIE,SAAS,CAACG,MAAnB,GACMkB,YAAY,KAAK,CADvB,GAEMA,YAAY,KAAKvB,KAJrB;AAMN2B,MAAAA,GAAG,EAAEJ,YANC;MAONvB,KAAK,EACDD,aAAa,CAACuB,YAAD,CAAb,GAA8BA,YAAY,CAACtB,KAA3C,GAAmDsB,YARjD;AAUNnB,MAAAA,UAAU,EAAEoB,YAVN;AAWNK,MAAAA,OAAO,EAAEpB,QAXH;MAYNqB,OAAO,EAAE9B,aAAa,CAACuB,YAAD,CAAb,GAA8BA,YAAY,CAACQ,IAA3C,GAAkD,IAAA;AAZrD,KAAV,CADJ,CAAA,CAAA;AAgBA,IAAA,OAAOT,KAAP,CAAA;AACH,GAlBA,EAkBE,EAlBF,CADL,CADJ,CAnBJ,CADJ,CAAA;AA6CH,CAAA;;AACDf,WAAW,CAACyB,WAAZ,GAA0B,aAA1B,CAAA;;AAUA,SAASN,SAAT,CAAmB;EAAEzB,KAAF;EAASG,UAAT;EAAqBuB,QAArB;EAA+BE,OAA/B;AAAwCC,EAAAA,OAAAA;AAAxC,CAAnB,EAAoF;AAChF,EAAA,MAAMG,IAAI,gBACNvB,KACgB,CAAAC,aADhB,CACgB,MADhB,EACgB;AAAA,IAAA,aAAA,EAAA,qBAAA;AACZI,IAAAA,SAAS,EAAE,qBAAyBY,IAAAA,QAAQ,GAAG,SAAH,GAAe,EAAhD,CADC;AAEZR,IAAAA,KAAK,EAAE;AAAEF,MAAAA,eAAe,EAAEhB,KAAAA;KAFd;AAGZ4B,IAAAA,OAAO,EAAE,MAAMA,OAAN,IAAMA,IAAAA,GAAAA,KAAAA,CAAAA,GAAAA,OAAO,CAAGzB,UAAH,CAAA;AAHV,GADhB,eAMIM,KAAA,CAAAC,aAAA,CAAA,MAAA,EAAA;AAAMI,IAAAA,SAAS,EAAC,wBAAA;AAAhB,GAAA,CANJ,CADJ,CAAA;AAWA,EAAA,OAAOe,OAAO,gBAAGpB,KAAA,CAAAC,aAAA,CAACuB,OAAD,EAAS;AAAAC,IAAAA,OAAO,EAAEL,OAAAA;AAAT,GAAT,EAA4BG,IAA5B,CAAH,GAAiDA,IAA/D,CAAA;AACH,CAAA;;AACDP,SAAS,CAACM,WAAV,GAAwB,WAAxB;;;;"}