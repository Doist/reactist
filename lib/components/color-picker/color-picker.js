"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("react"),t=require("classnames"),r=require("../deprecated-dropdown/dropdown.js"),o=require("../../tooltip/tooltip.js");function l(e){return e&&"object"==typeof e&&"default"in e?e:{default:e}}function n(e){if(e&&e.__esModule)return e;var t=Object.create(null);return e&&Object.keys(e).forEach((function(r){if("default"!==r){var o=Object.getOwnPropertyDescriptor(e,r);Object.defineProperty(t,r,o.get?o:{enumerable:!0,get:function(){return e[r]}})}})),t.default=e,t}var c=n(e),a=l(t);const i=["#606060","#4A90E2","#03B3B2","#008299","#82BA00","#D24726","#AC193D","#DC4FAD","#3BD5FB","#74E8D3","#FFCC00","#FB886E","#CCCCCC"],s=e=>"string"!=typeof e;function u({color:e=0,small:t,onChange:o,colorList:l=i}){return c.createElement(r.Dropdown.Box,{right:!0,className:"reactist_color_picker"},c.createElement(r.Dropdown.Trigger,null,(()=>{const r=((e,t)=>e[t>=e.length?0:t])(l,e);return c.createElement("span",{className:a.default("color_trigger",{small:t}),style:{backgroundColor:s(r)?r.color:r}},c.createElement("span",{className:"color_trigger--inner_ring"}))})()),c.createElement(r.Dropdown.Body,null,c.createElement("div",{className:"color_options"},l.reduce((t,r,n)=>(t.push(c.createElement(d,{isActive:e>=l.length?0===n:n===e,key:n,color:s(r)?r.color:r,colorIndex:n,onClick:o,tooltip:s(r)?r.name:null})),t),[]))))}function d({color:e,colorIndex:t,isActive:r,onClick:l,tooltip:n}){const a=c.createElement("span",{"data-testid":"reactist-color-item",className:"reactist color_item"+(r?" active":""),style:{backgroundColor:e},onClick:()=>null==l?void 0:l(t)},c.createElement("span",{className:"color_item--inner_ring"}));return n?c.createElement(o.Tooltip,{content:n},a):a}u.displayName="ColorPicker",d.displayName="ColorItem",exports.COLORS=i,exports.ColorItem=d,exports.ColorPicker=u;
//# sourceMappingURL=color-picker.js.map
