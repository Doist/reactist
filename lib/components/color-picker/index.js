"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("./color-picker.js");exports.COLORS=e.COLORS,exports.ColorItem=e.ColorItem,exports.default=e.ColorPicker;
//# sourceMappingURL=index.js.map
