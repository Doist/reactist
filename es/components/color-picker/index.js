import { ColorPicker } from './color-picker.js'
export { COLORS, ColorItem, ColorPicker as default } from './color-picker.js'
//# sourceMappingURL=index.js.map
