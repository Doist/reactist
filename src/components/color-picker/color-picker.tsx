import './color-picker.less'

import classnames from 'classnames'

import { Tooltip } from '../../tooltip'
import DeprecatedDropdown from '../deprecated-dropdown'

import type { ReactNode } from 'react'

type NamedColor = { name: string; color: string }

const COLORS = [
    '#606060',
    '#4A90E2',
    '#03B3B2',
    '#008299',
    '#82BA00',
    '#D24726',
    '#AC193D',
    '#DC4FAD',
    '#3BD5FB',
    '#74E8D3',
    '#FFCC00',
    '#FB886E',
    '#CCCCCC',
]

const _isNamedColor = (color: string | NamedColor | undefined): color is NamedColor =>
    typeof color !== 'string'

const _getColor = (colorList: (string | NamedColor)[], colorIndex: number) => {
    const index = colorIndex >= colorList.length ? 0 : colorIndex
    return colorList[index]
}

type Props = {
    small?: boolean
    color?: number
    onChange?: (color: number) => void
    colorList?: (string | NamedColor)[]
}

function ColorPicker({ color = 0, small, onChange, colorList = COLORS }: Props) {
    return (
        <DeprecatedDropdown.Box right className="reactist_color_picker">
            <DeprecatedDropdown.Trigger>
                {(() => {
                    const backgroundColor = _getColor(colorList, color)

                    return (
                        <span
                            className={classnames('color_trigger', { small })}
                            style={{
                                backgroundColor: _isNamedColor(backgroundColor)
                                    ? backgroundColor.color
                                    : backgroundColor,
                            }}
                        >
                            <span className="color_trigger--inner_ring" />
                        </span>
                    )
                })()}
            </DeprecatedDropdown.Trigger>
            <DeprecatedDropdown.Body>
                <div className="color_options">
                    {colorList.reduce<ReactNode[]>((items, currentColor, currentIndex) => {
                        items.push(
                            <ColorItem
                                isActive={
                                    color >= colorList.length
                                        ? currentIndex === 0
                                        : currentIndex === color
                                }
                                key={currentIndex}
                                color={
                                    _isNamedColor(currentColor) ? currentColor.color : currentColor
                                }
                                colorIndex={currentIndex}
                                onClick={onChange}
                                tooltip={_isNamedColor(currentColor) ? currentColor.name : null}
                            />,
                        )
                        return items
                    }, [])}
                </div>
            </DeprecatedDropdown.Body>
        </DeprecatedDropdown.Box>
    )
}
ColorPicker.displayName = 'ColorPicker'

type ColorItemProps = {
    color: string
    colorIndex: number
    isActive?: boolean
    onClick?: (colorIndex: number) => void
    tooltip?: ReactNode
}

function ColorItem({ color, colorIndex, isActive, onClick, tooltip }: ColorItemProps) {
    const item = (
        <span
            data-testid="reactist-color-item"
            className={'reactist color_item' + (isActive ? ' active' : '')}
            style={{ backgroundColor: color }}
            onClick={() => onClick?.(colorIndex)}
        >
            <span className="color_item--inner_ring" />
        </span>
    )

    return tooltip ? <Tooltip content={tooltip}>{item}</Tooltip> : item
}
ColorItem.displayName = 'ColorItem'

export { ColorItem, ColorPicker, COLORS }
