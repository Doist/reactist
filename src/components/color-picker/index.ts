import { ColorItem, ColorPicker, COLORS } from './color-picker'

export default ColorPicker
export { ColorItem, COLORS }
