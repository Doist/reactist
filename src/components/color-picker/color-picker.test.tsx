import React from 'react'
import { shallow } from 'enzyme'

import { ColorPicker, ColorItem } from './color-picker'

describe('ColorPicker', () => {
    it('renders without crashing', () => {
        const colorPicker = shallow(getColorPicker())
        expect(colorPicker).toMatchSnapshot()
    })

    it('renders with custom colorList', () => {
        const colorPicker = shallow(getColorPicker({ colorList: ['red', 'green', '#0000FF'] }))
        expect(colorPicker).toMatchSnapshot()
    })

    describe('ColorItem', () => {
        it('renders given color and does nothing when clicked without specified onClick handler', () => {
            const colorItem = shallow(getColorItem())
            colorItem.simulate('click')
            expect(colorItem).toMatchSnapshot()
        })

        it('adds active class when prop is supplied', () => {
            const colorItem = shallow(
                getColorItem({ isActive: true, colorIndex: 5, color: '#fff' }),
            )
            expect(colorItem.hasClass('active')).toBe(true)
        })

        it('calls onClick after clicking it', () => {
            const onClickSpy = jest.fn()
            const colorItem = shallow(
                getColorItem({
                    colorIndex: 5,
                    onClick: onClickSpy,
                    color: '#fff',
                }),
            )
            colorItem.simulate('click')
            expect(onClickSpy).toHaveBeenLastCalledWith(5)
        })
    })

    // Helpers ================================================================
    function getColorPicker(props?: React.ComponentProps<typeof ColorPicker>) {
        return <ColorPicker {...props} />
    }

    function getColorItem(props?: React.ComponentProps<typeof ColorItem>) {
        return <ColorItem color="#606060" colorIndex={0} {...props} />
    }
})
