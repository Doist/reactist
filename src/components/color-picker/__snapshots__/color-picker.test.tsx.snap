// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker ColorItem renders given color and does nothing when clicked without specified onClick handler 1`] = `
<div>
  <span
    class="reactist color_item"
    data-testid="reactist-color-item"
    style="background-color: rgb(96, 96, 96);"
  >
    <span
      class="color_item--inner_ring"
    />
  </span>
</div>
`;

exports[`ColorPicker renders with custom colorList 1`] = `
<div>
  <div
    class="reactist_dropdown reactist_color_picker"
    style="display: inline-block;"
  >
    <button
      aria-disabled="false"
      class="reactist_button trigger"
      type="button"
    >
      <span
        class="color_trigger"
        style="background-color: red;"
      >
        <span
          class="color_trigger--inner_ring"
        />
      </span>
    </button>
    <div
      class="body_wrapper with_arrow bottom"
      style="position: relative;"
    >
      <div
        class="body"
        id="reactist-dropdown-body"
        style="position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="color_options"
        >
          <span
            class="reactist color_item active"
            data-testid="reactist-color-item"
            style="background-color: red;"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: green;"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(0, 0, 255);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ColorPicker renders without crashing 1`] = `
<div>
  <div
    class="reactist_dropdown reactist_color_picker"
    style="display: inline-block;"
  >
    <button
      aria-disabled="false"
      class="reactist_button trigger"
      type="button"
    >
      <span
        class="color_trigger"
        style="background-color: rgb(96, 96, 96);"
      >
        <span
          class="color_trigger--inner_ring"
        />
      </span>
    </button>
    <div
      class="body_wrapper with_arrow bottom"
      style="position: relative;"
    >
      <div
        class="body"
        id="reactist-dropdown-body"
        style="position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="color_options"
        >
          <span
            class="reactist color_item active"
            data-testid="reactist-color-item"
            style="background-color: rgb(96, 96, 96);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(74, 144, 226);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(3, 179, 178);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(0, 130, 153);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(130, 186, 0);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(210, 71, 38);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(172, 25, 61);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(220, 79, 173);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(59, 213, 251);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(116, 232, 211);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(255, 204, 0);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(251, 136, 110);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
          <span
            class="reactist color_item"
            data-testid="reactist-color-item"
            style="background-color: rgb(204, 204, 204);"
          >
            <span
              class="color_item--inner_ring"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
